esp-link/cgiwifi.c

```diff
@@ -524,114 +729,233 @@ static char *wifiWarn[] = { 0,
+    
+    return len;
 }
 
 int ICACHE_FLASH_ATTR cgiWiFiConnStatus(HttpdConnData *connData) {
-  char buff[1024];
-  int len;
-
-  if (connData->conn==NULL) return HTTPD_CGI_DONE; // Connection aborted. Clean up.
-  jsonHeader(connData, 200);
-
-  len = os_sprintf(buff, "{");
-  len += printWifiInfo(buff+len);
-  len += os_sprintf(buff+len, ", ");
-
-  if (wifiReason != 0) {
-    len += os_sprintf(buff+len, "\"reason\": \"%s\", ", wifiGetReason());
-  }
-
+    char buff[1024];
+    int len;
+    
+    if (connData->conn==NULL) return HTTPD_CGI_DONE; // Connection aborted. Clean up.
+    jsonHeader(connData, 200);
+    
+    len = os_sprintf(buff, "{");
+    len += printWifiInfo(buff+len);
+    len += os_sprintf(buff+len, ", ");
+    
+    if (wifiReason != 0) {
+        len += os_sprintf(buff+len, "\"reason\": \"%s\", ", wifiGetReason());
+    }
+    
 #if 0
-  // commented out 'cause often the client that requested the change can't get a request in to
-  // find out that it succeeded. Better to just wait the std 15 seconds...
-  int st=wifi_station_get_connect_status();
-  if (st == STATION_GOT_IP) {
-    if (wifi_get_opmode() != 1) {
-      // Reset into AP-only mode sooner.
-      os_timer_disarm(&resetTimer);
-      os_timer_setfn(&resetTimer, resetTimerCb, NULL);
-      os_timer_arm(&resetTimer, 1000, 0);
-    }
-  }
+    // commented out 'cause often the client that requested the change can't get a request in to
+    // find out that it succeeded. Better to just wait the std 15 seconds...
+    int st=wifi_station_get_connect_status();
+    if (st == STATION_GOT_IP) {
+        if (wifi_get_opmode() != 1) {
+            // Reset into AP-only mode sooner.
+            os_timer_disarm(&resetTimer);
+            os_timer_setfn(&resetTimer, resetTimerCb, NULL);
+            os_timer_arm(&resetTimer, 1000, 0);
+        }
+    }
 #endif
-
-  len += os_sprintf(buff+len, "\"x\":0}\n");
-  //DBG("  -> %s\n", buff);
-  httpdSend(connData, buff, len);
-  return HTTPD_CGI_DONE;
+    
+    len += os_sprintf(buff+len, "\"x\":0}\n");
+    DBG("  -> %s\n", buff);
+    httpdSend(connData, buff, len);
+    return HTTPD_CGI_DONE;
 }
 
 // Cgi to return various Wifi information
 int ICACHE_FLASH_ATTR cgiWifiInfo(HttpdConnData *connData) {
-  char buff[1024];
-
-  if (connData->conn==NULL) return HTTPD_CGI_DONE; // Connection aborted. Clean up.
-
-  os_strcpy(buff, "{");
-  printWifiInfo(buff+1);
-  os_strcat(buff, "}");
+    char buff[1024];
+    
+    if (connData->conn==NULL) return HTTPD_CGI_DONE; // Connection aborted. Clean up.
+    
+    os_strcpy(buff, "{");
+    printWifiInfo(buff+1);
+    os_strcat(buff, "}");
+    
+    jsonHeader(connData, 200);
+    httpdSend(connData, buff, -1);
+    return HTTPD_CGI_DONE;
+}
 
-  jsonHeader(connData, 200);
-  httpdSend(connData, buff, -1);
-  return HTTPD_CGI_DONE;
+// Check string againt invalid characters
+int ICACHE_FLASH_ATTR checkString(char *str){
+    unsigned char strStripped[50];
```

Comment:
It's just because I am new to git-hub and as you said reading diff with new indentation was terrible, I used this function in an other contest. I have changed this to act as expected